pkg/controller/instance/instance_controller.go

```diff
@@ -187,8 +195,14 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 	// get the scheduled plan
 	plan, uid := scheduledPlan(instance, ov)
 	if plan == "" {
-		log.Printf("InstanceController: Nothing to do, no plan scheduled for instance %s/%s", instance.Namespace, instance.Name)
-		return reconcile.Result{}, nil
+		// no plan is running, we still need to make sure the readiness property is up to date
+		err := setReadinessOnInstance(instance, r.Client)
+		if err != nil {
+			log.Printf("InstanceController: Error when computing readiness for %s/%s: %v", instance.Namespace, instance.Name, err)
+			return reconcile.Result{}, err
+		}
+		err = updateInstance(instance, oldInstance, r.Client)
```

Comment:
Not yours, but maybe it makes sense to tackle this as part of this PR: How about creating a function `updateInstanceStatus` and refactor our existing code to better distinguish between updates to the spec vs updates to the status? Though I'm not sure how much work that is for the existing code, so feel free to ignore 😄

Reply:
hmm could you be more specific? Because `updateInstance` already tries to do the detection of what needs to be updated already.

Reply:
And by that it's implicit if either the spec or the status will be updated. My idea is to make it explicit. I.e., have a `updateInstanceStatus` funtion that only updates the status and a `updateInstanceSpec` function that only updates the spec. The callers should know which one to call. In this case it would be `updateInstanceStatus`.
Though, with `updateInstance` being a private function that is called just 3 times in `instance_controller.go`, it's okay to keep things as they are right now.

Reply:
yeah, I guess in this case we can call explicitly, but in the other use cases you would have to replicate all the comparison logic outside that method because you don't know if you need to update both simply because we're also updating Spec even though we should not be :) (controller should ideally never update spec)

Reply:
Looks like more work than anticipated. Let's revisit this later 😄